src/main/groovy/org/labkey/gradle/plugin/ModuleResources.groovy

```diff
@@ -41,10 +41,13 @@ class ModuleResources
                             Provider<Set<ResolvedArtifactResult>> artifacts = config.getIncoming().getArtifacts().getResolvedArtifacts();
                             task.getArtifactIds().set(artifacts.map(new WriteDependenciesFile.IdExtractor()))
                     }
+                    task.externalDependencies.set(project.extensions.findByType(ModuleExtension.class).getExternalDependencies())
+                    task.onlyIf {
+                        return !task.externalDependencies.get().isEmpty()
+                    }
                 } catch (UnknownDomainObjectException ignore) {
 
                 }
```

Comment:
Would an `UnknownDomainObjectException` cause this task to run when it shouldn't since `task.onlyIf {...}` won't be reached?
Should the `onlyIf` just be baked into `WriteDependenciesFile`, similar to `InstallRPackage`?

Reply:
Technically, that's two questions. 😄, but they are both good questions, so I'll let it slide. 

Effectively, it won't happen anymore that the exception is thrown since the configuration is created more broadly, but still a good idea to check. I had assumed Gradle wouldn't like the `onlyIf` inside the class, but that's cleaner, so I'll try it.

Reply:
There is no escape from off-by-one errors 😰